### PR TITLE
loosen up username and identity name validation

### PR DESCRIPTION
**Pull request by stlaz**
Allows to include ':' in usernames if prepended by 'b64'.
This is to allow creating identities/users whose IdP-side
identifier contains either ':' or '/'. These are base64-encoded
and the result is appended after the 'b64:' tag in the object
name.

cc @sttts @deads2k

### CODE REVIEW

**Comment by stlaz** [comment]
/hold
contains fake bump

**Comment by stlaz** [comment]
/retest
irrelevant

File: pkg/user/apis/user/validation/validation.go [hunk 1]
```diff
@@ -18,13 +18,20 @@ func ValidateIdentityName(name string, _ bool) []string {
 	}
 
 	parts := strings.Split(name, ":")
+	var providerIndex, usernameIndex = 0, 1
 	if len(parts) != 2 {
-		return []string{`must be in the format <providerName>:<providerUserName>`}
+		// allow identities in the form "<providerName>:b64:<base64-encoded username>"
+		if len(parts) == 3 && parts[1] == "b64" {
+			providerIndex++
```
Comment (sttts) [inline pkg/user/apis/user/validation/validation.go:25]: why providerIndex++ ?

**Comment by stlaz** [comment]
/hold cancel
no longer contains fake bump

File: pkg/user/apis/user/validation/validation.go [hunk 2]
```diff
@@ -18,13 +18,19 @@ func ValidateIdentityName(name string, _ bool) []string {
 	}
 
 	parts := strings.Split(name, ":")
+	usernameIndex := 1
 	if len(parts) != 2 {
-		return []string{`must be in the format <providerName>:<providerUserName>`}
+		// allow identities in the form "<providerName>:b64:<base64-encoded username>"
+		if len(parts) == 3 && parts[1] == "b64" {
```
Comment (sttts) [inline pkg/user/apis/user/validation/validation.go:24]: would be easier to read through
```
if len(parts) == 3 && parts[1] == "b64" {
 ...
} else if len(parts) != 2 { 
 ...
}
```

**Comment by sttts** [comment]
/retest
/lgtm
/approve

Feel free to fix the nit as the long tests have to rerun anyway.

**Pull request merged**